akka-epl/src/main/scala/com/epl/akka_v1/HttpClient.scala

```diff
@@ -0,0 +1,85 @@
+package com.epl.akka_v1
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.marshalling.Marshal
+import akka.http.scaladsl.model.headers.BasicHttpCredentials
+import akka.http.scaladsl.model.{HttpMethods, HttpRequest, RequestEntity, headers}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import com.epl.akka.DocumentType
+
+import scala.concurrent.Future
+
+object HttpClient {
+
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = system.dispatcher
+
+  final val config = system.settings.config
+
+  private val authorization = headers.Authorization(BasicHttpCredentials(config.getString("cloudant.username"), config.getString("cloudant.password")))
+
+
+  /**
+    * POST call to Cloudant database API
+    * @param jsonString
+    * @return
+    */
+  def post(jsonString: String): Future[String] = {
+    val responseEntity = for {
+      request <- Marshal(jsonString).to[RequestEntity]
+      response <- Http().singleRequest(HttpRequest(
+        method = HttpMethods.POST,
+        uri = config.getString("cloudant.post_url"),
+        headers = List(authorization),
+        entity = request))
+      entity <- Unmarshal(response.entity).to[ByteString]
+    } yield entity
+
+    responseEntity.mapTo[String]
+
+  }
+
+
+  /**
+    * GET call to Cloudant database API
+    * @param documentType
+    * @return
+    */
+  def get(documentType: DocumentType.Documenttype): Future[String] = {
+    val responseEntity = for {
+      response <- Http().singleRequest(HttpRequest(
+        method = HttpMethods.GET,
+        uri = getUrl(documentType),
+        headers = List(authorization)))
+      entity <- Unmarshal(response.entity).to[ByteString]
+    } yield entity
+
+    responseEntity.mapTo[String]
+
+  }
+
+
+  /**
+    * Get URI based on the type of document requested.
+    * @param documentType
+    * @return
+    */
+  def getUrl(documentType: DocumentType.Documenttype): String ={
+    var url: String = null
+    if(documentType.equals(DocumentType.TeamTable)){
+      url = config.getString("cloudant.get_tables_url")
+    }else if(documentType.equals(DocumentType.Results)){
+      url = config.getString("cloudant.get_results_url")
+    }else{
+      url = config.getString("cloudant.get_fixtures_url")
```

Comment:
Reading config is somewhat costly, so it makes sense to rather read these once and cache the values in fields of the class when constructing the instance, just like you do with the auth header.

Reply:
Would creating a immutable map like below would work?
```
class AppConfig(implicit val system: ActorSystem,
                implicit val materializer: ActorMaterializer ) {
  private var config = scala.collection.immutable.Map[String, String]()
  def load(): Unit ={
    val appConfig = ConfigFactory.load();
    config+= ("username" -> appConfig.getString("cloudant.username"))
   ...
  }
  def getConfig(key: String): String ={
    config(key)
  }
}
```

Reply:
Yes, you could use a map, but I think just using fields are more clear and allow you to keep separate types with the values rather than either forcing all configurable values to be strings or 

Note that you should not load the configuration yourself, instead you should use it from the actor context, so:

```Scala
class AppConfig(implicit system: ActorSystem, materializer: ActorMaterializer ) {
   // gives us a config just containing the things inside cloudant { ... } in the config file
  private def config = system.settings.config.getConfig("cloudant")
  // these are strings, as for example
  private val username = config.getString("username")
  private val resultsUrl = config.getString("get_results_url")
  // just an example, a boolean setting
  private val useHttps = config.getBoolean("use-https")
... the rest of the class ... 
```